lighthouse-core/scripts/lantern/evaluate-results.js

```diff
@@ -0,0 +1,153 @@
+    const rankDiffAsPercent = rankDiff / entries.length;
+
+    rankErrors.push(rankDiffAsPercent);
+    percentErrors.push(diffAsPercent);
+    entry = {...entry, expected, actual, diff, rankDiff, rankDiffAsPercent, metric};
+    if (diffAsPercent < GOOD_ABSOLUTE_THRESHOLD || rankDiffAsPercent < GOOD_RANK_THRESHOLD) {
+      good.push(entry);
+    } else if (diffAsPercent < OK_ABSOLUTE_THRESHOLD) {
+      ok.push(entry);
+    } else bad.push(entry);
+  }
+
+  if (lanternMetric.includes('roughEstimate')) {
+    totalGood.push(...good);
+    totalOk.push(...ok);
+    totalBad.push(...bad);
+  }
+
+  const MAPE = Math.round(percentErrors.reduce((x, y) => x + y) / percentErrors.length * 1000) / 10;
+  const rank = Math.round(rankErrors.reduce((x, y) => x + y) / rankErrors.length * 1000) / 10;
+  const buckets = `${good.length}/${ok.length}/${bad.length}`;
+  console.log(
+    metric.padEnd(30),
+    lanternMetric.padEnd(25),
+    `${rank}%`.padEnd(12),
+    `${MAPE}%`.padEnd(10),
+    buckets.padEnd(15)
+  );
+}
+
+console.log('----    Metric Stats    ----');
+console.log(
+  'metric'.padEnd(30),
+  'estimate'.padEnd(25),
+  'rank error'.padEnd(12),
+  'MAPE'.padEnd(10),
+  'Good/OK/Bad'.padEnd(15)
+);
+evaluateBuckets('firstContentfulPaint', 'optimisticFCP');
+evaluateBuckets('firstContentfulPaint', 'pessimisticFCP');
+evaluateBuckets('firstContentfulPaint', 'roughEstimateOfFCP');
+
+evaluateBuckets('firstMeaningfulPaint', 'optimisticFMP');
+evaluateBuckets('firstMeaningfulPaint', 'pessimisticFMP');
+evaluateBuckets('firstMeaningfulPaint', 'roughEstimateOfFMP');
+
+evaluateBuckets('timeToFirstInteractive', 'optimisticTTFCPUI');
+evaluateBuckets('timeToFirstInteractive', 'pessimisticTTFCPUI');
+evaluateBuckets('timeToFirstInteractive', 'roughEstimateOfTTFCPUI');
+
+evaluateBuckets('timeToConsistentlyInteractive', 'optimisticTTI');
+evaluateBuckets('timeToConsistentlyInteractive', 'pessimisticTTI');
+evaluateBuckets('timeToConsistentlyInteractive', 'roughEstimateOfTTI');
+
+evaluateBuckets('speedIndex', 'optimisticSI');
+evaluateBuckets('speedIndex', 'pessimisticSI');
+evaluateBuckets('speedIndex', 'roughEstimateOfSI');
+
+const total = totalGood.length + totalOk.length + totalBad.length;
+console.log('\n----    Summary Stats    ----');
+console.log(`Good: ${Math.round(totalGood.length / total * 100)}%`);
+console.log(`OK: ${Math.round(totalOk.length / total * 100)}%`);
+console.log(`Bad: ${Math.round(totalBad.length / total * 100)}%`);
+
+console.log('\n----    Worst10 Sites    ----');
+for (const entry of totalBad.sort((a, b) => b.rankDiff - a.rankDiff).slice(0, 10)) {
+  console.log(
+    entry.actual < entry.expected ? 'underestimated' : 'overestimated',
+    entry.metric,
+    'by',
+    Math.round(entry.diff),
+    'on',
+    entry.url
+  );
+}
+
+/**
+ * @typedef LanternSiteDefinition
+ * @property {string} url
+ * @property {string} tracePath
+ * @property {string} devtoolsLogPath
+ * @property {*} lantern
```

Comment:
whats the def for this? is it just that it's long?

Reply:
fixed